presto-orc/src/main/java/com/facebook/presto/orc/reader/SliceDictionarySelectiveReader.java

```diff
@@ -271,6 +282,26 @@ private int readWithFilter(int[] positions, int positionCount)
         return streamPosition;
     }
 
+    private byte evaluateFilter(int position, int index, int length)
+    {
+        if (filter.testLength(length)) {
+            int currentPosLength = dictionaryBlock.getSliceLength(index);
+            Slice data = dictionaryBlock.getSlice(index, 0, currentPosLength);
+            if (isCharType && length != currentPosLength) {
```

Comment:
Is it always currentPosLength < length <=  maxCodePointCount when `length != currentPosLength` ? Could you explain a bit why we pad it to maxCodePointCount not length?

Reply:
if we have a char(10), then the maxCodePoint is 10. If we have data with ending spaces "abc  " then they are trimmed and stored. The index length here is differing for hive/presto writers. if its hive its writing the trimmed length 3 and presto is writing 5. so that the check where I compare 3 to 5 and append spaces. maxCodePoint  >= currentLenght and length. currenLength and length are always equal expect when written by hive/presto which have spaces